WeScan/Extensions/CIImage+Utils.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  CIImage+Utils.swift
+//  WeScan
+//
+//  Created by Julian Schiavo on 14/11/2018.
+//  Copyright © 2018 WeTransfer. All rights reserved.
+//
+
+import CoreImage
+import Foundation
+
+extension CIImage {
+    func applyingAdaptiveThreshold() -> UIImage? {
+        guard let colorKernel = CIColorKernel(source:
+            "kernel vec4 color(__sample pixel, float inputEdgeO, float inputEdge1)" +
+                "{" +
+                "    float luma = dot(pixel.rgb, vec3(0.2126, 0.7152, 0.0722));" +
+                "    float threshold = smoothstep(inputEdgeO, inputEdge1, luma);" +
+                "    return vec4(threshold, threshold, threshold, 1.0);" +
+            "}"
```

Comment:
This looks pretty hard to maintain, but I guess there is no really other way of doing this, isn't it? Is there a way to make this type safe?

Reply:
I'm afraid it seems CoreImage is just an old framework with poor Swift code:
> Core Image also not able to rely on large parts of Swift's type safety, so you need to be careful when using it because the compiler won't help you as much as you're used to.